src/scripts/main.js

```diff
@@ -1,11 +1,22 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    element.addEventListener(eventName, (e) => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    });
+  });
 }
```

Comment:
Also, the event listener added in waitFor function is never removed. This can lead to memory leaks if the element is removed from the DOM. Consider removing the event listener once it's been invoked.